devel/icu/Portfile

```diff
@@ -62,6 +63,8 @@ if { [vercmp ${version} 59] < 0 } {
                             size    34655619
 } else {
     compiler.cxx_standard   2011
+    # see https://trac.macports.org/ticket/59391
+    compiler.blacklist-append {clang < 700}
```

Comment:
Does specifying the C standard, as well as that for C++, not work ? i.e. `compiler.c_standard ` ?

Reply:
Unfortunately, no.
Based on the [release notes](https://releases.llvm.org/3.1/docs/ClangReleaseNotes.html#cchanges), the MacPorts base assumes C11 support [starts in Clang version 318.0.61](https://github.com/macports/macports-base/blob/785ca255c8da2608ae2046b9f369ea791fc0e76c/src/port1.0/portconfigure.tcl#L783).
`max_align_t`, however seems to be missing.

Reply:
It seems to have been considered a very problematic part of the standard by the clang devs. The link Ken posted in the ticket said that "There is an
effort to discourage any further use of this construct as a consequence [of the issues]."

Reply:
I have no strong feelings about his, but there was a [fair amount of discussion](https://github.com/macports/macports-base/pull/88) of where to put the boundary for C99 compliant compilers and what constituted full compliance.
There might be similar concerns for C11.
Even if we made the change in base, however, it would not be available for some time.
Meanwhile, [icu is broken on 10.9](https://trac.macports.org/ticket/59391).

Reply:
Do not get me wrong, I am not suggesting delaying deploying the fixes here in order to wait for a possible future base update (that may or may not happen...). By all means go ahead with the local blacklisting. I am just saying, in my view, requesting a C11 standards compliant compiler should result in base blacklisting anything which does not fully support that standard, either officially or 'unofficially'.... So in my view base is not stringent enough in this regard.

Reply:
b.t.w. I don't have time to go through the PR you linked, so I have not ready all the discussion there. But for me its simple. A compiler implements a standard fully, or it does not. There is no middle ground.

Reply:
Then by your definition, no compiler we have access to implements C99.

Reply:
There is always "a middle ground" with compilers that only partially support any particular standard, often being released before the standard even gets finalized:
* https://en.cppreference.com/w/cpp/compiler_support

Obviously, one piece of software may only use the most trivial feature from C++17, making it unsuitable to use the compiler that support just C++11, but perfectly fine to use the one that only implements super basic support. On on the other hand another piece of software may be using almost every single feature from the language, and only the most advance support would cut it ... There are flags like `-std=c++0x` as a predecessor of `c++11` etc.

Some build systems (including `cmake` and `autoconf`) would let you test for individual features of the standard, but doing that for MacPorts and specifying the exact list of features needed to get that software compiled is definitely an overkill.

What we could do is specify a standard, make a very strict compiler selection, while letting maintainers optionally add and older compiler to the allowed list. Doing it the opposite (listing compilers with incomplete features) would likely be too much work.